pkg/storagesvc/stowClient.go

```diff
@@ -62,6 +62,43 @@ var (
 	ErrWritingFileIntoResponse = errors.New("unable to copy item into http response")
 )
 
+func getContainer(loc stow.Location, containerName string, cursor string) (stow.Container, error) {
+	// use location.Containers to find containers that match the prefix (container name)
+	cons, cursorNew, err := loc.Containers(containerName, cursor, 1)
+	if err != nil {
+		return nil, err
+	}
+	var con stow.Container
+	for _, v := range cons {
+		c, err := loc.Container(v.ID())
+		if err != nil {
+			return nil, err
+		}
+		if c.Name() == containerName {
+			con = cons[0]
+			break
+		}
+	}
+	if con == nil && !stow.IsCursorEnd(cursorNew) {
+		_, err := getContainer(loc, containerName, cursorNew)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return con, nil
+}
+
+func getOrCreateContainer(loc stow.Location, containerName string, cursor string) (stow.Container, error) {
+	con, err := loc.CreateContainer(containerName)
+	if err != nil && (os.IsExist(err) || strings.Contains(err.Error(), "BucketAlreadyOwnedByYou")) {
+		con, err = getContainer(loc, containerName, stow.CursorStart)
+	}
+	if con == nil {
```

Comment:
We should have an additional condition here.

```
if con == nil && err == nil {
}
```

If err is not nil, ideally, we should wrap error, so that we don't lose previously captured error.

```
fmt.Errorf("Storage container %s not found: %w", containerName, err)
```

Also, in the below error container name is missing.